src/v/rpc/types.h

```diff
@@ -204,6 +204,24 @@ class streaming_context {
     std::vector<ssx::semaphore_units> _reservations;
 };
 
+/**
+ * @brief Bundles the method_id and assoicated method name.
+ */
+struct method_info {
+    /**
+     * The name of this method, which should be used only for logging and other
+     * diagnosis, and should not be assumed unique or unchanged across commits.
+     * The pointer-to string must have indefinite lifetime (e.g., a string
+     * literal is probably the way to go).
+     */
+    const char* name;
```

Comment:
nit: we may make it `string_view`

Reply:
Good idea. I was also thinking we should try to enforce the "indefinite lifetime" aspect along the lines of:

https://stackoverflow.com/questions/69124293/is-it-possible-in-c-to-enforce-a-string-literal-function-argument
and
https://stackoverflow.com/questions/74922986/c20-concept-matching-string-literal

Reply:
@mmaslankaprv - Hmm, in this case I choose a single 8-byte value explicitly since I want to keep this object cheap to pass by value: structures up to 16 bytes (which meet certain other requirements wrt POD/triviality or whatever) can be passed in  registers, but string view itself is 16 bytes + 4 for id = 20 (actually ends up 24 due to alignment), so the structure has to be passed in-memory on the stack.

So this was deliberate: I wanted this to be cheap to pass, almost as cheap as the uint32_t it is replacing. What's the primary downside you see? Outputting this string is very slightly more expensive during formatting since we need a `strlen`.